ambry-commons/src/test/java/com.github.ambry.commons/BlobIdTest.java

```diff
@@ -13,119 +13,447 @@
 import java.nio.ByteBuffer;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collections;
 import java.util.List;
+import java.util.Random;
 import java.util.UUID;
 import org.apache.commons.codec.binary.Base64;
+import org.junit.Before;
 import org.junit.Test;
 
+import static com.github.ambry.clustermap.ClusterMapUtils.*;
+import static com.github.ambry.commons.BlobId.*;
+import static com.github.ambry.utils.Utils.*;
 import static org.junit.Assert.*;
 
 
+/**
+ * Unit tests for {@link BlobId}. Using {@link BlobIdV2} builds a {@link BlobId}
+ * in v2 equivalent to setting {@link BlobId#CURRENT_VERSION} to be {@link BlobId#BLOB_ID_V2}.
+ */
 public class BlobIdTest {
+  private static final Random random = new Random();
+  private static final List<Short> versions = Arrays.asList(new Short[]{BLOB_ID_V1, BLOB_ID_V2});
+  private byte referenceFlag;
+  private byte referenceDatacenterId;
+  private short referenceAccountId;
+  private short referenceContainerId;
+  private ClusterMap referenceClusterMap;
+  private PartitionId referencePartitionId;
+
+  /**
+   * Initialization before each unit test.
+   * @throws Exception Any unexpected exception.
+   */
+  @Before
+  public void init() throws Exception {
+    byte[] bytes = new byte[2];
+    referenceClusterMap = new MockClusterMap();
+    random.nextBytes(bytes);
+    referenceFlag = bytes[0];
+    referenceDatacenterId = bytes[0];
+    referenceAccountId = getRandomShort(random);
+    referenceContainerId = getRandomShort(random);
+    referencePartitionId = referenceClusterMap.getWritablePartitionIds().get(0);
+  }
+
+  /**
+   * Tests building blobId in both {@link BlobId#BLOB_ID_V1} and {@link BlobId#BLOB_ID_V2}. The expected values
+   * for {@code flag}, {@code datacenterId}, {@code accountId}, {@code containerId}, and {@code partitionId} are
+   * listed below:
+   * <pre>
+   * Version
+   * 1            always default values except partitionId
+   * 2            values passed in as arguments
+   * </pre>
+   * @throws Exception Any unexpected exception.
+   */
   @Test
-  public void basicTest() throws Exception {
-    final long id = 99;
-    PartitionId partitionId = new MockPartitionId(id, Collections.EMPTY_LIST, 0);
-    BlobId blobId = new BlobId(partitionId);
+  public void testBuildBlobId() throws Exception {
+    for (Short version : versions) {
+      buildBlobIdAndAssert(version, referenceFlag, referenceDatacenterId, referenceAccountId, referenceContainerId,
+          referencePartitionId);
+    }
+  }
 
-    assertEquals(blobId.getPartition(), partitionId);
-    System.out.println("Blob Id toString: " + blobId);
-    System.out.println("Blob id sizeInBytes: " + blobId.toString().length());
+  /**
+   * Tests deserialization from a composed blobId string (not from deserialization of a blobId in both version
+   * 1 and 2.
+   * @throws Exception Any unexpected exception.
+   */
+  @Test
+  public void testDeserialization() throws Exception {
```

Comment:
I don't understand the need of this test. We have the next one which tests serialization and deserialization if I am not wrong. Or does this tests something more than the next one ?